app/models/authenticator/httpd.rb

```diff
@@ -48,5 +52,45 @@ def update_user_attributes(user, _username, identity)
       user.last_name  = user_attrs[:lastname]
       user.email      = user_attrs[:email] unless user_attrs[:email].blank?
     end
+
+    def get_user_details_by_userid(username)
```

Comment:
Unless this is being written to conform to an existing pattern, can you remove teh `get` prefix, as it's generally avoided in Ruby code. e.g. here it could just be `user_details_by_userid`